File: apis/users-service/src/repositories/UsersRepository/interface.ts
import User from '@entities/User';

export default interface UsersRepository {
  find(id: string): Promise<User | undefined>;
  findByEmail(email: string): Promise<User | undefined>;
  store(user: User): Promise<void>;
  update(user: User): Promise<void>;
  delete(id: string): Promise<void>;
}

File: apis/users-musics-service/internal/repositories/usersMusicsRepository/interface.go
package usersMusicsRepository

import e "users-musics-service/internal/entities"

type IUsersMusicsRepository interface {
	Store(request StoreUserMusicRequest) error
	Update(request UpdateUserMusicRequest) error
	Delete(request DeleteUserMusicRequest) error
	Find(request FindUserMusicRequest) (*e.UserMusic, error)
	FindLastUpdated(request FindLastUpdatedRequest) ([]e.UserMusic, error)
	FindMostViews(request FindMostViewsRequest) ([]e.UserMusic, error)
}
